sonar-php-plugin/src/main/java/org/sonar/plugins/php/reports/ExternalReportWildcardProvider.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.php.reports;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.Version;
+import org.sonarsource.analyzer.commons.FileProvider;
+
+public final class ExternalReportWildcardProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExternalReportWildcardProvider.class);
+  private static final int MINIMUM_MAJOR_SUPPORTED_VERSION = 7;
+  private static final int MINIMUM_MINOR_SUPPORTED_VERSION = 2;
+
+  private ExternalReportWildcardProvider() {
+    // util class
+  }
+
+  public static List<File> getReportFiles(SensorContext context, String externalReportsProperty) {
+    String[] reportPaths = context.config().getStringArray(externalReportsProperty);
+    if (reportPaths.length == 0) {
+      return Collections.emptyList();
+    }
+
+    var minumumVersion = Version.create(MINIMUM_MAJOR_SUPPORTED_VERSION, MINIMUM_MINOR_SUPPORTED_VERSION);
+    boolean externalIssuesSupported = context.runtime().getApiVersion().isGreaterThanOrEqual(minumumVersion);
+    if (!externalIssuesSupported) {
+      LOG.error("Import of external issues requires SonarQube 7.2 or greater.");
+      return Collections.emptyList();
+    }
+
+    List<File> result = new ArrayList<>();
+    for (String reportPath : reportPaths) {
+      var reports = getIOFiles(context.fileSystem().baseDir(), reportPath);
+      result.addAll(reports);
+    }
+
+    return result;
+  }
+
+  private static List<File> getIOFiles(File baseDir, String reportPath) {
+    var fileProvider = new FileProvider(baseDir, reportPath);
```

Comment:
The idea is the same as in https://github.com/SonarSource/sonar-iac/pull/1481, however I found this API in a similar case in sonar-js: https://github.com/SonarSource/SonarJS/blob/master/sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/lcov/CoverageSensor.java#L99